torch/utils/backend_registration.py

```diff
@@ -205,12 +205,11 @@ def wrap_storage_to(self, device=None, non_blocking=False):
         # For sparse storage, custom need to extend the implementation by themselves.
         if self.is_sparse:
             raise RuntimeError(f"Can not support a sparse storage move to {custom_backend_name} backend")
-        # create untyped_storage and copy data
-        untyped_storage = torch.UntypedStorage(
-            self.size(), device=torch.device(f'{custom_backend_name}:{device_idx}')
-        )
-        untyped_storage.copy_(self, non_blocking)
-        return untyped_storage
+        # copy data by tensor
```

Comment:
1. non_blocking is a problem, I will fix.

2. About Creating storage by tensor:
In some scenarios, we get a subclass `CustomStorageImpl` by extend `StorageImpl`, `CustomStorageImpl` has many unique properties of open device.
Use `torch.UntypedStorage` method to create only `StorageImpl` but not `CustomStorageImpl`.
By creating tensors, open device can control the storage creation process and create `CustomStorageImpl`.

Reply:
For 2, I would expect the UntypedStorage constructor to do the right thing for the device?
The c++ storage constructor must already be doing the right thing otherwise even the empty() call you use here wouldn't work?